codalab/model/bundle_model.py

```diff
@@ -830,19 +770,45 @@ def resume_bundle(self, bundle, bundle_update, row, user_id, worker_id, connecti
 
         return True
 
-    def finalize_bundle(self, bundle, user_id, exitcode, failure_message, connection):
+    def transition_bundle_worker_offline(self, bundle):
```

Comment:
Finalize moved below `transition_bundle_worker_offline`.
The code for `transition_bundle_worker_offline` is the same as the old `set_bundle_offline`, which just moved down here from above. None of the code diff below is changed code.